src/genre/genre.controller.ts

```diff
@@ -0,0 +1,46 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { GenreService } from './genre.service';
+import { CreateGenreDto } from './dto/create-genre.dto';
+import { UpdateGenreDto } from './dto/update-genre.dto';
+import { Character } from 'src/entity/character.entity';
+
+@Controller('genre')
+export class GenreController {
+  constructor(private readonly genreService: GenreService) {}
+
+  @Post()
+  create(@Body() createGenreDto: CreateGenreDto) {
+    return this.genreService.create(createGenreDto);
+  }
+
+  @Get()
+  findAll() {
+    return this.genreService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.genreService.findOne(+id);
+  }
+
+  @Patch(':id')
+  update(@Param('id') id: string, @Body() updateGenreDto: UpdateGenreDto) {
+    return this.genreService.update(+id, updateGenreDto);
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.genreService.remove(+id);
+  }
+
+  @Get('movie/:movieId')
```

Comment:
This URI structure doesn't make sense. URIs usually follow a hierarchical structure. If you want to get a movie's genres using the movie ID the URI should be `/movie/:id/genre`